### DISCUSSION TITLE
Symfony constraint DateTimeType - validate that selected date is today

### DISCUSSION THREAD

**Discussion by PhilippBartels** [symfony/symfony#43356] [Q&A] [answered]
Hello people,

I've a form with 2 Fields with the `DateTimeType`. Here is my `buildForm` method:
```
public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('begin', DateTimeType::class, ['widget' => 'single_text', 'required' => true, "label" => "Start", 'data' => new \DateTime()])
            ->add('end', DateTimeType::class, ['widget' => 'single_text','required' => true, "label" => "Ende", 'data' => new \DateTime()]);
    }
```

**I need to validate with a constraint that the selected _Date in both fields is today_ and that _end is not greater than begin_**.
Can someone help me to build the fitting constraint rules?

Thanks!

**Comment by PhilippBartels** [comment, marked as answer]
I found the solution. Its possible to pass the time as string to GreaterThanOrEqual and LessThanOrEqual

```
->add('begin', DateTimeType::class, ['widget' => 'single_text', 'required' => true, "label" => "Start", 'data' => new \DateTime(), 'constraints' => [
                new GreaterThanOrEqual($dates[0]),
                new LessThanOrEqual($dates[1])
            ]])
```

**Comment by benjaminmal** [comment]
You can even use [PHP date and time formats](https://www.php.net/manual/en/datetime.formats.php) in the constraint as explain here [GreaterThanOrEqual#comparing-dates](https://symfony.com/doc/current/reference/constraints/GreaterThanOrEqual.html#comparing-dates) and use [`propertyPath`](https://symfony.com/doc/current/reference/constraints/GreaterThanOrEqual.html#propertypath) parameter to compare 2 properties:
```php
public function buildForm(FormBuilderInterface $builder, array $options): void
{
    $builder
        ->add('begin', DateTimeType::class, [
            'widget' => 'single_text',
            'required' => true,
            'label' => 'Start',
            'data' => new \DateTime(),
            'constraints' => [
                new GreaterThanOrEqual('today'),
            ],
        ])
        ->add('end', DateTimeType::class, [
            'widget' => 'single_text',
            'required' => true,
            'label' => 'End',
            'data' => new \DateTime(),
            'constraints' => [
                new LessThanOrEqual(propertyPath: 'parent.all[begin].data'), # or just 'begin' if you set your constraint with annotation in your entity
            ],
        ])
    ;
}
```

**Reply by PhilippBartels** [reply]
> You can even use [PHP date and time formats](https://www.php.net/manual/en/datetime.formats.php) in the constraint as explain here [GreaterThanOrEqual#comparing-dates](https://symfony.com/doc/current/reference/constraints/GreaterThanOrEqual.html#comparing-dates) and use [`propertyPath`](https://symfony.com/doc/current/reference/constraints/GreaterThanOrEqual.html#propertypath) parameter to compare 2 properties:
> 
> ```
> public function buildForm(FormBuilderInterface $builder, array $options): void
> {
>     $builder
>         ->add('begin', DateTimeType::class, [
>             'widget' => 'single_text',
>             'required' => true,
>             'label' => 'Start',
>             'data' => new \DateTime(),
>             'constraints' => [
>                 new GreaterThanOrEqual('today'),
>             ],
>         ])
>         ->add('end', DateTimeType::class, [
>             'widget' => 'single_text',
>             'required' => true,
>             'label' => 'End',
>             'data' => new \DateTime(),
>             'constraints' => [
>                 new LessThanOrEqual(propertyPath: 'parent.all[begin].data'), # or just 'begin' if you set your constraint with annotation in your entity
>             ],
>         ])
>     ;
> }
> ```

Wow, this is really cool. Thank you - this is working perfectly. For other users with the same problem, here my constraint configuration to ensure that the second date field is not before (in time) the first selection and between fixed date times (in my case first and last second of today). 
```

'constraints' => [
                new GreaterThanOrEqual($dates[0]),
                new LessThanOrEqual($dates[1]),
                new GreaterThan(propertyPath: 'parent.all[begin].data'),
            ]
```
